java/vector/src/main/java/org/apache/arrow/vector/NullableIntervalDayVector.java

```diff
@@ -0,0 +1,363 @@
+
+   public NullableIntervalDayVector(String name, BufferAllocator allocator) {
+      this(name, FieldType.nullable(Types.MinorType.INTERVALDAY.getType()),
+              allocator);
+   }
+
+   public NullableIntervalDayVector(String name, FieldType fieldType, BufferAllocator allocator) {
+      super(name, allocator, fieldType, TYPE_WIDTH);
+      reader = new IntervalDayReaderImpl(NullableIntervalDayVector.this);
+   }
+
+   @Override
+   protected org.slf4j.Logger getLogger() {
+      return logger;
+   }
+
+   @Override
+   public FieldReader getReader(){
+      return reader;
+   }
+
+   @Override
+   public Types.MinorType getMinorType() {
+      return Types.MinorType.INTERVALDAY;
+   }
+
+
+   /******************************************************************
+    *                                                                *
+    *          vector value retrieval methods                        *
+    *                                                                *
+    ******************************************************************/
+
+
+   /**
+    * Get the element at the given index from the vector.
+    *
+    * @param index   position of element
+    * @return element at given index
+    */
+   public ArrowBuf get(int index) throws IllegalStateException {
+      if(isSet(index) == 0) {
+         return null;
+      }
+      return valueBuffer.slice(index * TYPE_WIDTH, TYPE_WIDTH);
+   }
+
+   /**
+    * Get the element at the given index from the vector and
+    * sets the state in holder. If element at given index
+    * is null, holder.isSet will be zero.
+    *
+    * @param index   position of element
+    */
+   public void get(int index, NullableIntervalDayHolder holder){
+      if(isSet(index) == 0) {
+         holder.isSet = 0;
+         return;
+      }
+      final int startIndex = index * TYPE_WIDTH;
+      holder.isSet = 1;
+      holder.days = valueBuffer.getInt(startIndex);
+      holder.milliseconds = valueBuffer.getInt(startIndex + 4);
+   }
+
+   /**
+    * Same as {@link #get(int)}.
+    *
+    * @param index   position of element
+    * @return element at given index
+    */
+   public Period getObject(int index) {
+      if (isSet(index) == 0) {
+         return null;
+      } else {
+         final int startIndex = index * TYPE_WIDTH;
+         final int days = valueBuffer.getInt(startIndex);
+         final int milliseconds = valueBuffer.getInt(startIndex + 4);
+         final Period p = new Period();
+         return p.plusDays(days).plusMillis(milliseconds);
+      }
+   }
+
+   public StringBuilder getAsStringBuilder(int index) {
+      if (isSet(index) == 0) {
+         return null;
+      }else{
+         return getAsStringBuilderHelper(index);
+      }
+   }
+
+   private StringBuilder getAsStringBuilderHelper(int index) {
+      final int startIndex = index * 8;
+
+      final int  days = valueBuffer.getInt(startIndex);
+      int millis = valueBuffer.getInt(startIndex + 4);
```

Comment:
maybe add a `MILLISECOND_OFFSET = 4` and use that here so it gives some context on what the calculation is for

Reply:
Done.